packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/TaskNavButtons/components/ExpertOptions/ExpertOptionsContainer.spec.js

```diff
@@ -0,0 +1,108 @@
+import React from 'react'
+import { mount, shallow } from 'enzyme'
+import sinon from 'sinon'
+import { Grommet } from 'grommet'
+import theme from '@zooniverse/grommet-theme'
+import ExpertOptionsContainer from './ExpertOptionsContainer'
+import ExpertOptions from './ExpertOptions'
+
+describe('TaskNavButtons > Component > ExpertOptionsContainer', function () {
+  let wrapper, useContextMock
+
+  const mockStoreWithoutUser = {
+    classifierStore: {
+      authClient: {
+        checkCurrent: sinon.stub().callsFake(() => Promise.resolve(null))
+      }
+    }
+  }
+
+  const mockStoreWithAdminUser = {
+    classifierStore: {
+      authClient: {
+        checkCurrent: sinon.stub().callsFake(() => Promise.resolve({ admin: true }))
+      }
+    }
+  }
+
+  const mockStoreWithStandardUser = {
+    classifierStore: {
+      authClient: {
+        checkCurrent: sinon.stub().callsFake(() => Promise.resolve({ admin: false }))
+      }
+    }
+  }
+
+  describe('smoke test', function () {
+    before(function () {
+      useContextMock = sinon.stub(React, 'useContext').callsFake(() => mockStoreWithoutUser)
+      wrapper = mount(
+        <ExpertOptionsContainer />
+      )
+    })
+
+    after(function () {
+      useContextMock.restore()
+    })
+
+    it('should render without crashing', function () {
+      expect(wrapper).to.be.ok()
+    })
+  })
+
+  describe('when there is no user', function () {
+    before(function () {
+      useContextMock = sinon.stub(React, 'useContext').callsFake(() => mockStoreWithoutUser)
+      wrapper = mount(
+        <ExpertOptionsContainer />
+      )
+    })
+
+    after(function () {
+      useContextMock.restore()
+    })
+
+    it('should render null', function () {
+      expect(wrapper.html()).to.be.empty()
+    })
+  })
+
+  describe('when there is a standard user', function () {
+    before(function () {
+      useContextMock = sinon.stub(React, 'useContext').callsFake(() => mockStoreWithStandardUser)
+      wrapper = mount(
+        <ExpertOptionsContainer />
+      )
+    }, {
+      wrappingComponent: Grommet,
+      wrappingComponentProps: { theme }
+    })
+
+    after(function () {
+      useContextMock.restore()
+    })
+
+    it('should not render ExpertOptions', function () {
+      console.log(wrapper.debug())
+      expect(wrapper.find(ExpertOptions)).to.have.lengthOf(0)
+    })
+  })
+
+  xdescribe('when there is an admin user', function () {
```

Comment:
Does this test still need to be fixed?

Reply:
For reference, this test is skipped because it throws an error inside the `DropButton` that's rendered for the toggle.

```
TypeError: Cannot read property 'length' of undefined
    at phash (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/styled-components/dist/styled-components.cjs.js:683:13)
    at ComponentStyle.generateAndInjectStyles (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/styled-components/dist/styled-components.cjs.js:1271:25)
    at useInjectedStyle (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/styled-components/dist/styled-components.cjs.js:1497:120)
    at useStyledComponentImpl (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/styled-components/dist/styled-components.cjs.js:1525:28)
    at StyledButton (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/styled-components/dist/styled-components.cjs.js:1592:12)
```

My best guess at what might be wrong here is that `styled-components` uses context, but this test is mocking `React.useContext`.

Reply:
One way to get around this would be to add an error boundary around the `DropButton`. It's not ideal, but I think it would allow you to test the container.